src/algorithm/hnswlib/BlockManager.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstring>
+#include <deque>
+#include <functional>
+#include <mutex>
+
+#include "../../default_allocator.h"
+#include "StreamReader.h"
+#include "StreamWriter.h"
+
+namespace hnswlib {
+
+class BlockManager {
+public:
+    BlockManager(size_t max_elements,
+                 size_t size_data_per_element,
+                 size_t block_size_limit,
+                 vsag::Allocator* allocator);
+
+    ~BlockManager();
+
+    char*
+    getElementPtr(size_t index, size_t offset);
```

Comment:
code style: public function(interface)'s name should start with a captain letter

Reply:
done